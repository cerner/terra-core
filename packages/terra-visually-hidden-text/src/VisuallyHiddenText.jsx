import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames/bind';
import styles from './VisuallyHiddenText.module.scss';

const cx = classNames.bind(styles);

const propTypes = {
  /**
   * Callback ref to pass into the dom element. This is useful when using terra-visually-hidden-text as an aria-live container.
   */
  refCallback: PropTypes.func,
  /**
   * Text to be read to the screen reader
   */
  text: PropTypes.string,
};

const defaultProps = {
  text: undefined,
  refCallback: undefined,
};

const VisuallyHiddenText = ({ refCallback, text, ...customProps }) => {
  const VisuallyHiddenTextClassNames = cx([
    'visually-hidden-text',
    customProps.className,
  ]);

  return (<span ref={refCallback} {...customProps} className={VisuallyHiddenTextClassNames}>{text}</span>);
};

VisuallyHiddenText.propTypes = propTypes;
VisuallyHiddenText.defaultProps = defaultProps;

export default VisuallyHiddenText;
