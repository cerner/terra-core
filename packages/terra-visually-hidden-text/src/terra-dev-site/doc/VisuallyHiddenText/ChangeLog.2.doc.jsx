import React from 'react';
import DocTemplate from 'terra-doc-template';
import ChangeLog from '../../../../CHANGELOG.md';

const DocPage = () => (
  <DocTemplate
    packageName="visually-hidden-text"
    srcPath="https://github.com/cerner/terra-core/tree/master/packages/visually-hidden-text"
    readme={ChangeLog}
  />
);

export default DocPage;
