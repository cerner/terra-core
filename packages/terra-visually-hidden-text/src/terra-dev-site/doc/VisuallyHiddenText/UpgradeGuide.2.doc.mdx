import { Badge } from 'terra-visually-hidden-text/package.json?dev-site-package';

<Badge />

# Terra Visually Hidden Text Upgrade Guide

## Changes from 1.0 to 2.0

### Changes to CSS Custom Properties

The following transitive dependency had breaking changes. See its upgrade guide for more information.
* terra-doc-template
