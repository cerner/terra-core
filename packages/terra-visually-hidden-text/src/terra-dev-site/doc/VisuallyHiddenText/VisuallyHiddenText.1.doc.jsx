/* eslint-disable import/no-webpack-loader-syntax, import/first, import/extensions, import/no-duplicates, import/no-unresolved */
import React from 'react';
import DocTemplate from 'terra-doc-template';
import ReadMe from '../../../../docs/README.md';
import { name } from '../../../../package.json';

// Component Source
import VisuallyHiddenTextSrc from '!raw-loader!../../../../src/VisuallyHiddenText';

// Example Files
import DefaultVisuallyHiddenText from '../example/DefaultVisuallyHiddenText';
import DefaultVisuallyHiddenTextSrc from '!raw-loader!../../../../src/terra-dev-site/doc/example/DefaultVisuallyHiddenText.jsx';

const DocPage = () => (
  <DocTemplate
    packageName={name}
    readme={ReadMe}
    srcPath={`https://github.com/cerner/terra-core/tree/master/packages/${name}`}
    examples={[
      {
        title: 'VisuallyHiddenText with empty string',
        example: <DefaultVisuallyHiddenText />,
        source: DefaultVisuallyHiddenTextSrc,
      },
    ]}
    propsTables={[
      {
        componentName: 'VisuallyHiddenText',
        componentSrc: VisuallyHiddenTextSrc,
      },
    ]}
  />
);

export default DocPage;
