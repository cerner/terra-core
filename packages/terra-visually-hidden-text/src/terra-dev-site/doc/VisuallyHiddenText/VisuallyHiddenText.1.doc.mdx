import { Badge } from 'terra-visually-hidden-text/package.json?dev-site-package';

import DefaultVisuallyHiddenText from '../example/DefaultVisuallyHiddenText?dev-site-example';
import RefCallbackVisuallyHiddenText from '../example/RefCallbackVisuallyHiddenText?dev-site-example';

import VisuallyHiddenTextPropsTable from 'terra-visually-hidden-text/src/VisuallyHiddenText?dev-site-props-table';

<Badge />

# Terra Visually Hidden Text

Visually Hidden Text is a component designed for screen readers that renders text on the dom, but is not visible to the UI. Some components such as badge will have a visual indication to note their hierarchy on the page, but that indication is lost when a screen reader is used on the web page. Visually Hidden text would allow your application to provide that additional context to a screen reader.

Another reason for this component is when you want to produce more semantic markup for a screen reader only experience. Some various examples include:

- You may want to provide a visually hidden header to allow for better readability.
- You may need to provide different instructions for visual users vs non visual users.
- div nodes have a difficult understanding the aria-label attributes as well.

In these instances, it's recommended to use visually hidden text.

## Getting Started

- Install with [npmjs](https://www.npmjs.com):
  - `npm install terra-visually-hidden-text`

<!-- AUTO-GENERATED-CONTENT:START Peer Dependencies -->
## Peer Dependencies

This component requires the following peer dependencies be installed in your app for the component to properly function.

| Peer Dependency | Version |
|-|-|
| react | ^16.8.5 |
| react-dom | ^16.8.5 |

<!-- AUTO-GENERATED-CONTENT:END -->

## Usage
```jsx
import VisuallyHiddenText from '../../../VisuallyHiddenText';
```

## Component Features

 * [Cross-Browser Support](https://engineering.cerner.com/terra-ui/#/getting-started/terra-ui/features)

[View component source code](https://github.com/cerner/terra-core/tree/master/packages/terra-visually-hidden-text)

## Examples
<DefaultVisuallyHiddenText title="VisuallyHiddenText with empty string" />
<RefCallbackVisuallyHiddenText title="VisuallyHiddenText with ref Callback" />

## Visually Hidden Text Props
<VisuallyHiddenTextPropsTable />
