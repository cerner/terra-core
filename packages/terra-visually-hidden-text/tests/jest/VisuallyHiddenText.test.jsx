import React from 'react';
import VisuallyHiddenText from '../../src/VisuallyHiddenText';

describe('VisuallyHiddenText', () => {
  const defaultRender = <VisuallyHiddenText text="This is screen reader only text" />;

  // Snapshot Tests
  it('should render a default component', () => {
    const wrapper = shallow(defaultRender);
    expect(wrapper).toMatchSnapshot();
  });

  it('should pass in refCallback as the ref prop of the input element', () => {
    const refCallback = jest.fn();
    const wrapper = mount(<VisuallyHiddenText refCallback={refCallback} />);
    expect(refCallback).toBeCalled();
    expect(wrapper).toMatchSnapshot();
  });
});
