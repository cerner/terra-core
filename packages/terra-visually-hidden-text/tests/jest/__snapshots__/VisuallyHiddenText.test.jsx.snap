// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VisuallyHiddenText should render a default component 1`] = `
<span
  className="visually-hidden-text"
>
  This is screen reader only text
</span>
`;
